homework/zadanie5_false_sharing.cpp

```diff
@@ -40,58 +40,58 @@ int main(){
     std::cout<< "\n---------------------------------------default size: \n\n";
 
     auto start = std::chrono::high_resolution_clock::now();
-    fill(array, 0, size, fill_it_up);
+    fill(array, size, 0, 1, fill_it_up);
     auto stop = std::chrono::high_resolution_clock::now();
-    std::cout<< "Time of function -> fill(array_without_padding): " << std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count() << "\n\n";
+    std::cout<< "Time of fun -> fill(array_without_padding):          " << std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count() << "\n\n";
 
     start = std::chrono::high_resolution_clock::now();
-    fill(array64, 0, size, fill_it_up);
+    std::thread t([&](){fill(array, size, 0, 2, fill_it_up);});
+    std::thread t1([&](){fill(array, size, 1, 2, fill_it_up);});
+    t.join();
+    t1.join();
     stop = std::chrono::high_resolution_clock::now();
-    std::cout<< "Time of function -> fill(array_with_padding): " << std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count() << "\n\n";
+    std::cout<< "Time of parallel fun -> fill(array_without_padding): " << std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count() << "\n\n";
 
     start = std::chrono::high_resolution_clock::now();
-    std::thread t([&](){fill(array, 0, size-1, fill_it_up);});
-    std::thread t1([&](){fill(array, size/2, size, fill_it_up);});
-    t.join();
-    t1.join();
+    fill(array64, size, 0, 1, fill_it_up);
     stop = std::chrono::high_resolution_clock::now();
-    std::cout<< "Time of parallel function -> fill(array_without_padding): " << std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count() << "\n\n";
+    std::cout<< "Time of fun -> fill(array_with_padding):             " << std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count() << "\n\n";
 
     start = std::chrono::high_resolution_clock::now();
-    std::thread t3([&](){fill(array64, 0, size-1, fill_it_up);});
-    std::thread t4([&](){fill(array64, size/2, size, fill_it_up);});
+    std::thread t3([&](){fill(array64, size, 0, 2, fill_it_up);});
+    std::thread t4([&](){fill(array64, size, 1, 2, fill_it_up);});
```

Comment:
a tak w ogóle to lambdę można pominąć.
Nie wiem jak to wpłynie na rezultaty, ale zobacz

`std::thread t3(fill, array64, size, 0, 2, fill_it_up);`

Reply:
Na poczatku opakowalem ta funkcje w lambde, zeby wyjsc z bledu:), bo nie moge przekazac szablonowej funkcji wraz z argumentami do watku, probowalem na ponizsze sposoby, ale dalej bez skutku:

1) std::thread t(fill, array, size, 0, 2, fill_it_up);

error: zadanie5_false_sharing.cpp:50:54: error: no matching function for call to ‘std::thread::thread(<unresolved overloaded function type>, Int [size], int&, int, int, int&)’
   50 |     std::thread t(fill, array, size, 0, 2, fill_it_up);

2) std::thread t(&fill<Int>(array, size, 0, 2, fill_it_up)

error: zadanie5_false_sharing.cpp: In function ‘int main()’:
zadanie5_false_sharing.cpp:50:59: error: lvalue required as unary ‘&’ operand
   50 |     std::thread t(&fill<Int>(array, size, 0, 2, fill_it_up));

rowniez probowalem opakowac szablonowa funkcje w klase, ale rowniez nie bylo oczekiwanych rezultatow, dalej szukam rozwiazania